nibabel/cifti2/parse_cifti2.py

```diff
@@ -147,7 +147,6 @@ def __init__(self, dataobj, affine, header=None,
         if self.cifti_img is None:
             raise ValueError('Nifti2 header does not contain a CIFTI2 '
                              'extension')
-        self.cifti_img.data = self.get_data()
```

Comment:
Or could be `self.cifti_img.dataobj = self.dataobj`.  This doesn't load the data, but keeps it as an array proxy.